src/plan.rs

```diff
@@ -73,6 +73,52 @@ const MAX_RADIX4_BITS: u32 = 16; // largest size to consider radix 4 an option i
 const MAX_RADER_PRIME_FACTOR: usize = 23; // don't use Raders if the inner fft length has prime factor larger than this
 const MIN_BLUESTEIN_MIXED_RADIX_LEN: usize = 90; // only use mixed radix for the inner fft of Bluestein if length is larger than this
 
+#[derive(Debug, std::cmp::PartialEq, Clone)]
+#[allow(dead_code)]
+pub enum Recipe {
+    DFT(usize),
+    MixedRadix {
+        left_fft: Box<Recipe>,
+        right_fft: Box<Recipe>,
+    },
+    GoodThomasAlgorithm {
+        left_fft: Box<Recipe>,
+        right_fft: Box<Recipe>,
+    },
+    MixedRadixSmall {
+        left_fft: Box<Recipe>,
+        right_fft: Box<Recipe>,
+    },
+    GoodThomasAlgorithmSmall {
+        left_fft: Box<Recipe>,
+        right_fft: Box<Recipe>,
+    },
+    RadersAlgorithm {
+        inner_fft: Box<Recipe>,
+    },
+    BluesteinsAlgorithm {
+        len: usize,
+        inner_fft: Box<Recipe>,
+    },
+    Radix4(usize),
+    Butterfly2,
```

Comment:
Does explicitly listing out all the butterfly sizes here help? If not, it seems like we could reduce some repetition by making this `Butterfly(usize)`

Reply:
I had it like that at first. The reason I switched to one enum variant per length wes that I didn't want to keep two completely separate lists of the butterflies. This way I still need two sets of matches, but if there is a typo it's more likely the compiler will complain. But the Butterfly(len) is shorter and looks nicer.
Both ways work and I don't have any strong preference.

Reply:
I like the argument that offloads more of the work of remembering onto the compiler.